util/src/org/commcare/util/cli/CliCommand.java

```diff
@@ -0,0 +1,87 @@
+package org.commcare.util.cli;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.commcare.util.CommCareConfigEngine;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+public abstract class CliCommand {
+    protected final String commandName;
+    protected final String helpTextDescription;
+    protected final String positionalArgsHelpText;
+
+    protected String[] args;
+    protected Options options;
+    protected CommandLine cmd;
+
+    public CliCommand(String commandName, String helpTextDescription, String positionalArgsHelpText) {
+        options = getOptions();
+        this.commandName = commandName;
+        this.helpTextDescription = helpTextDescription;
+        this.positionalArgsHelpText = positionalArgsHelpText;
+    }
+
+    /*
+        :args: - all the args after (not including) commandName
+     */
+    public void parseArguments(String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        cmd = parser.parse(options, args);
+        this.args = cmd.getArgs();
+    }
+
+    public void checkHelpOption() {
+        if (cmd.hasOption("h")) {
+            printHelpText();
+            System.exit(0);
+        }
+    }
+
+    public void handle() {
+        checkHelpOption();
+    }
+
+    protected Options getOptions() {
+        Options options = new Options();
+        options.addOption(Option.builder("h")
+                .desc("Get a list of options")
+                .build());
+        return options;
+    }
+
+    protected void printHelpText() {
+        String usage = "java -jar commcare-cli.jar " + commandName + " " + positionalArgsHelpText;
+        String header = helpTextDescription + "\n";
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.setWidth(200);
+        formatter.printHelp(usage, header, options, "", true);
+    }
+
+    public String getHelpTextDescription() {
+        return helpTextDescription;
+    }
+
+    public String getCommandName() {
+        return commandName;
+    }
```

Comment:
you can safely make `final` fields public, which means you don't need getter methods like `getCommandName` or `getHelpTextDescription`. Of course making use of this pattern is a matter of taste.